fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/OnlineIndexer.java

```diff
@@ -1817,28 +1829,51 @@ private static void checkPositive(int value, String desc) {
      * A builder for the indexing policy.
      */
     public static class IndexingPolicy {
-        public static final IndexingPolicy DEFAULT = new IndexingPolicy();
         @Nullable private final String sourceIndex;
         @Nullable private final Object sourceIndexSubspaceKey; // overrides the sourceIndex
         private final boolean forbidRecordScan;
+        private DesiredAction ifDisabled;
+        private DesiredAction ifWriteOnly;
+        private DesiredAction ifMismatchPrev;
+        private DesiredAction ifReadable;
+
+        public enum DesiredAction {
+            ERROR,
+            REBUILD,
+            CONTINUE,
+        }
 
         /**
          * Build the index from a source index. Source index must be readable, idempotent, and fully cover the target index.
          * @param sourceIndex source index
          * @param sourceIndexSubspaceKey if non-null, overrides the sourceIndex param
          * @param forbidRecordScan forbid fallback to a by-records scan
+         * @param ifDisabled desired action if the existing index state is DISABLED
+         * @param ifWriteOnly desired action if the existing index state is WRITE_ONLY (i.e. partly built)
+         * @param ifMismatchPrev desired action if the index is partly built, but by a different method then currently requested
+         * @param ifReadable desired action if the existing index state is READABLE (i.e. already built)
          */
-        public IndexingPolicy(@Nullable String sourceIndex, @Nullable Object sourceIndexSubspaceKey, boolean forbidRecordScan) {
+        public IndexingPolicy(@Nullable String sourceIndex, @Nullable Object sourceIndexSubspaceKey, boolean forbidRecordScan,
+                              DesiredAction ifDisabled, DesiredAction ifWriteOnly, DesiredAction ifMismatchPrev, DesiredAction ifReadable) {
             this.sourceIndex = sourceIndex;
             this.forbidRecordScan = forbidRecordScan;
             this.sourceIndexSubspaceKey = sourceIndexSubspaceKey;
+            this.ifDisabled = ifDisabled;
+            this.ifWriteOnly = ifWriteOnly;
+            this.ifMismatchPrev = ifMismatchPrev;
+            this.ifReadable = ifReadable;
         }
 
         /**
          * Build a non-active object.
          */
         public IndexingPolicy() {
-            this(null, null, false);
+            this(null, null, false,
+                    DesiredAction.REBUILD, DesiredAction.CONTINUE, DesiredAction.CONTINUE, DesiredAction.CONTINUE);
+        }
+
+        public static IndexingPolicy defaultPolicy() {
+            return new IndexingPolicy();
```

Comment:
Should the default policy be a single object that gets cached (so that `defaultPolicy()` doesn't have to create a new object)? Does the object need to be immutable first before that can be accomplished?

Reply:
If the object is final, then probably yes. 
I'll change it back.

Reply:
Fixed